packages/app/src/app/pages/common/Modals/SubscriptionCancellation/SubscriptionCancellation.tsx

```diff
@@ -0,0 +1,232 @@
+      (teamUsage?.privateProjectsQuantity ?? 0) -
+      TEAM_FREE_LIMITS.private_repos;
+    if (restrictedRepositoriesCount > 0) {
+      return `${restrictedRepositoriesCount} restricted`;
+    }
+
+    return undefined;
+  };
+
+  return (
+    <Stack
+      css={{
+        position: 'relative',
+        overflow: 'hidden',
+        padding: '24px 32px 32px',
+      }}
+      direction="vertical"
+    >
+      <Stack justify="space-between">
+        <Text
+          as="h1"
+          css={{
+            margin: 0,
+            lineHeight: '28px',
+            letterSpacing: '-0.019em',
+          }}
+          size={19}
+          weight="normal"
+        >
+          You&apos;ll lose access to all Pro features
+        </Text>
+        <IconButton name="cross" title="Close" onClick={modalClosed} />
+      </Stack>
+      <Stack
+        as="ul"
+        css={{
+          listStyle: 'none',
+          margin: '24px 0 0',
+          padding: 0,
+        }}
+        direction="vertical"
+        gap={6}
+      >
+        {FEATURES.map(f => {
+          let badge = f.pill;
+          if (FEATURES_WITH_DYNAMIC_PILLS.includes(f.key)) {
+            badge = {
+              editors: getEditorsLabel(),
+              sandboxes: getSandboxesLabel(),
+              repos: getReposLabel(),
+            }[f.key];
+          }
+
+          return (
+            <Stack direction="vertical" key={f.key} gap={1}>
+              <Text
+                css={{
+                  textDecoration: 'line-through',
+                  color: '#999999',
+                }}
+                size={13}
+              >
+                {f.pro}
+              </Text>
+              <Stack gap={3}>
+                <Text size={13}>{f.free}</Text>
+                {badge ? <Badge variant="warning">{badge}</Badge> : null}
+              </Stack>
+            </Stack>
+          );
+        })}
+      </Stack>
+      <Stack
+        css={{
+          flexWrap: 'wrap',
+          marginLeft: 'auto',
+          marginRight: 0,
+          marginTop: '32px',
+        }}
+        gap={6}
+      >
+        <Button onClick={modalClosed} variant="link" autoWidth>
+          {hasActiveTeamTrial ? 'Continue trial' : 'Keep subscription'}
+        </Button>
+        {isPaddle ? (
+          <Button
+            css={{ padding: '0 32px' }}
+            loading={paddleLoading}
+            onClick={() => {
+              setPaddeLoading(true);
+              // TO DO: confirm event name.
+              track('Team Settings - Cancel paddle subscription', {
+                codesandbox: 'V1',
+                event_source: 'UI',
+              });
+
+              pro.cancelWorkspaceSubscription();
+            }}
+            autoWidth
+          >
+            I understand
+          </Button>
+        ) : (
+          <Button
+            css={{ padding: '0 32px' }}
+            loading={loadingCustomerPortal}
+            onClick={() => {
+              // TO DO: confirm event name.
```

Comment:
👀

Reply:
Confirmed on Slack and updated here: [74939d3](https://github.com/codesandbox/codesandbox-client/pull/7195/commits/74939d32f835c52a6388873e81a47ae8603e0f9c).